### PR TITLE
Using secrets

### PR DESCRIPTION
**Pull request by dannylamb**
This approach lets you use secrets for running in production.  For an environment variable that should be secret, instead of directly entering its value into config, replace it with `secret:/path/to/the/secret`.  This little init script will scan the container environment for variables that follow that pattern, read the contents of that file, and replace the variable's value with the file's contents.

There will be an analagous pull on isle-dc to demonstrate how to use secrets in the docker-compose file.  I'm trying to work this in such that if integrating with other things like etcd, vault or aws's secrets solution, that this won't get in the way. It's specifically for the env backend of confd and if you don't pattern your variables this way, nothing will change.

### CODE REVIEW

File: demo/Dockerfile [hunk 1]
```diff
@@ -22,7 +22,6 @@ RUN --mount=type=cache,target=/root/.composer/cache \
         drupal/facets:^1.3 \
         drupal/matomo:^1.7 \
         drupal/pdf:1.x-dev \
-        drupal/rdfui:^1.0-beta1 \
```
Comment (nigelgbanks) [inline demo/Dockerfile:25]: I assume this is fine, just thought I'd bring attention to it if it was intended for another branch or commit.

**Review by nigelgbanks — APPROVED** [review]

File: base/rootfs/etc/cont-init.d/00-load-secrets.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/with-contenv bash
+set -e
+
+# Read any secret files specified in environment variables.
+echo "$(env | grep '=secret:')" | while read line
```
Comment (noahwsmith) [inline base/rootfs/etc/cont-init.d/00-load-secrets.sh:5]: Could you put instructions into the README? It looks like you have to do a specific thing with a filename to make this work, and we shouldn't assume people will go read the code to figure it out.

Comment (noahwsmith) [reply base/rootfs/etc/cont-init.d/00-load-secrets.sh:5]: And can you add your SECRETfile as a template to this repo?

Comment (noahwsmith) [reply base/rootfs/etc/cont-init.d/00-load-secrets.sh:5]: If you scroll to "By .env technique" here you can see how to load a second .yml file by changing your COMPOSE_FILE declaration in your .env file https://stackoverflow.com/questions/55650342/import-docker-compose-file-in-another-compose-file

**Comment by dannylamb** [comment]
I've made an associated pull to isle-dc: https://github.com/Islandora-Devops/isle-dc/pull/123

I'll try to document it in the README as best I can here.

**Comment by dannylamb** [comment]
I decided it fit better in isle-dc than here.  I updated the PR to include a secrets chunk in the README.

File: base/rootfs/etc/cont-init.d/00-load-secrets.sh (hunk 2 above)
Comment (dannylamb) [reply base/rootfs/etc/cont-init.d/00-load-secrets.sh:5]: I put example secret files in https://github.com/Islandora-Devops/isle-dc/pull/123

**Comment by dannylamb** [comment]
@nigelgbanks Any chance we can get this in to help with https://github.com/Islandora-Devops/isle-dc/pull/126?

**Comment by g7morris** [comment]
@dannylamb per your request here https://github.com/Islandora-Devops/isle-dc/pull/126#issuecomment-726818269 I merged this PR

**Pull request merged**